### PR TITLE
docs: Add initial mTLS setup

### PR DESCRIPTION
**Pull request by guilhem-barthes**
Add a new page in the how-to section to describe how to set up mTLS

Signed-off-by: Guilhem Barthes <guilhem.barthes@owkin.com>

### CODE REVIEW

File: docs/source/operations/howto/mtls-setup.rst [hunk 1]
```diff
@@ -3,54 +3,56 @@ How to set up mTLS
...
-    This guide assumes that you already have followed the instructions to :ref:`set up TLS<operations/howto/orchestrator-deployment:Setup TLS>`.
+This guide assumes that you already have followed the instructions to :ref:`set up TLS<operations/howto/orchestrator-deployment:Setup TLS>`.
 
-Generate backend :abbr:`CSR (Certificate Signing Request)` and key
-==================================================================
+Generate backend Certificate Signing Request and signing key
+============================================================
 
-The first step is to generate the :abbr:`CSR (Certificate Signing Request)` for the :term:`organization`.
+The first step is to generate the Certificate Signing Request and a Signing key for the :term:`organization`.
```
Comment (guilhem-barthes) [inline docs/source/operations/howto/mtls-setup.rst:15]: Signing (in Signing key) is capitalized here but not in the title

File: docs/source/operations/howto/mtls-setup.rst [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+In this page we will describe how to set up mTLS communication between the backends and the orchestrator.
+In this scenario, the orchestrator will act as the certificate authority checking the certificates.
+These instructions have to be repeated for each backend.
+
+This guide assumes that you already have followed the instructions to :ref:`set up TLS<operations/howto/orchestrator-deployment:Setup TLS>`.
+
+Generate backend Certificate Signing Request and signing key
+============================================================
+
+The first step is to generate the Certificate Signing Request and a Signing key for the :term:`organization`.
```
Comment (guilhem-barthes) [inline docs/source/operations/howto/mtls-setup.rst:15]:
```suggestion
The first step is to generate the Certificate Signing Request and a signing key for the :term:`organization`.
```

File: docs/source/operations/howto/mtls-setup.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
+******************
+How to set up mTLS
+******************
+
+
+In this page we will describe how to set up mTLS communication between the backends and the orchestrator.
```
Comment (ThibaultFy) [inline docs/source/operations/howto/mtls-setup.rst:6]:
```suggestion
This page describes how to set up mTLS communication between the backends and the orchestrator.
```

File: docs/source/operations/howto/mtls-setup.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  It should be the same as the value you put in your ``values.yaml`` file for the key ``orchestrator.mspID``.
+| Replace ``HOSTNAME`` with the hostname of your substra backend.
+
+Then, you will need to send the file named ``ORGNAME.csr`` to the organization managing the orchestrator for them to sign your certificate.
+
+Signing the Substra backend certificate
+=======================================
+
+Now that you have the Certificate Signing Request from your backend in your orchestrator, you can sign it with the orchestrator certificate authority.
+You need to navigate to the directory where are located the files ``orchestrator-ca.crt`` and ``orchestrator-ca.key`` (created during :ref:`TLS setup<operations/howto/orchestrator-deployment:Setup TLS>`).
```
Comment (ThibaultFy) [inline docs/source/operations/howto/mtls-setup.rst:31]:
```suggestion
Now that you have the Certificate Signing Request from your backend in your orchestrator, you can sign it with the orchestrator certificate authority.
You need to navigate to the directory where the files ``orchestrator-ca.crt`` and ``orchestrator-ca.key`` are located (created during :ref:`TLS setup<operations/howto/orchestrator-deployment:Setup TLS>`).
```

File: docs/source/operations/howto/mtls-setup.rst [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+.. code:: bash
+
+   openssl req -newkey rsa:2048 -nodes -keyout ORGNAME.key -subj "/O=ORGNAME/CN=HOSTNAME" -out ORGNAME.csr
+
+| Replace ``ORGNAME`` with your :term:`organization` name.
+  It should be the same as the value you put in your ``values.yaml`` file for the key ``orchestrator.mspID``.
+| Replace ``HOSTNAME`` with the hostname of your substra backend.
+
+Then, you will need to send the file named ``ORGNAME.csr`` to the organization managing the orchestrator for them to sign your certificate.
```
Comment (ThibaultFy) [inline docs/source/operations/howto/mtls-setup.rst:25]:
```suggestion
Then, you need to send the file named ``ORGNAME.csr`` to the organization managing the orchestrator for them to sign your certificate.
```

File: docs/source/operations/howto/mtls-setup.rst [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+Signing the Substra backend certificate
+=======================================
+
+Now that you have the Certificate Signing Request from your backend in your orchestrator, you can sign it with the orchestrator certificate authority.
+You need to navigate to the directory where are located the files ``orchestrator-ca.crt`` and ``orchestrator-ca.key`` (created during :ref:`TLS setup<operations/howto/orchestrator-deployment:Setup TLS>`).
+
+Then you can sing the certificate with the following command:
```
Comment (ThibaultFy) [inline docs/source/operations/howto/mtls-setup.rst:33]:
```suggestion
You can sign the certificate with the following command:
```

File: docs/source/operations/howto/mtls-setup.rst [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+        enabled: true
+        cacert: orchestrator-cacert
+        mtls:
+            enabled: true
+            clientCertificate: orchestrator-client-cert
+
+Note that you need to have the orchestrator TLS enabled for this to work.
+
+Once your config file is updated, you can either redeploy the backend to apply the changes or continue following the backend deployment guide.
+The backend can be updated with the following command:
```
Comment (ThibaultFy) [inline docs/source/operations/howto/mtls-setup.rst:70]:
```suggestion
Once your config file is updated, you can either redeploy the backend to apply the changes or continue the backend deployment guide.
The backend can be updated with the following command:
```

File: docs/source/operations/howto/backend-deployment.rst [hunk 8]
```diff
@@ -106,6 +106,8 @@ To configure your values:
              enabled: true
              cacert: orchestrator-cacert
 
+#. Optional: If the orchestrator requires your backend to be authenticated by using mTLS follow :doc:`mtls-setup`.
```
Comment (ThibaultFy) [inline docs/source/operations/howto/backend-deployment.rst:109]: Can we change the "How to" files order to have this reference pointing to a previous link ?

File: docs/source/operations/howto/mtls-setup.rst (hunk 3 above)
Comment (guilhem-barthes) [reply docs/source/operations/howto/mtls-setup.rst]: Applied

File: docs/source/operations/howto/mtls-setup.rst (hunk 4 above)
Comment (guilhem-barthes) [reply docs/source/operations/howto/mtls-setup.rst]: Done!

File: docs/source/operations/howto/mtls-setup.rst (hunk 6 above)
Comment (guilhem-barthes) [reply docs/source/operations/howto/mtls-setup.rst]: Done!

File: docs/source/operations/howto/mtls-setup.rst (hunk 7 above)
Comment (guilhem-barthes) [reply docs/source/operations/howto/mtls-setup.rst]: Done!

File: docs/source/operations/howto/backend-deployment.rst (hunk 8 above)
Comment (guilhem-barthes) [reply docs/source/operations/howto/backend-deployment.rst:109]: So the order would be orchestrator > mTLS > backend ?

Comment (ThibaultFy) [reply docs/source/operations/howto/backend-deployment.rst:109]: Mmmh no sorry I'm must have been confused, the order seems fine this way...

Comment (ThibaultFy) [reply docs/source/operations/howto/backend-deployment.rst:109]: Maybe put the how by order of importance, no strong opinion on the order though

**Review by RomainGoussault — APPROVED** [review]

**Pull request merged**